homeassistant/components/brother/sensor.py

```diff
@@ -25,356 +28,351 @@
-        )
+    value: Callable[[BrotherSensors], StateType | datetime]
+    extra_state_attrs: Callable[[BrotherSensors], dict[str, Any]]
 
 
 @dataclass
-class BrotherSensorEntityDescription(SensorEntityDescription):
+class BrotherSensorEntityDescription(
+    SensorEntityDescription, BrotherSensorRequiredKeysMixin
+):
     """A class that describes sensor entities."""
 
-    entity_class: type[BrotherPrinterSensor] = BrotherPrinterSensor
-
 
 SENSOR_TYPES: tuple[BrotherSensorEntityDescription, ...] = (
     BrotherSensorEntityDescription(
-        key=ATTR_STATUS,
+        key="status",
         icon="mdi:printer",
         name="Status",
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.status,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_PAGE_COUNTER,
+        key="page_counter",
         icon="mdi:file-document-outline",
         name="Page counter",
         native_unit_of_measurement=UNIT_PAGES,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.page_counter,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_BW_COUNTER,
+        key="bw_counter",
         icon="mdi:file-document-outline",
         name="B/W counter",
         native_unit_of_measurement=UNIT_PAGES,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.bw_counter,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_COLOR_COUNTER,
+        key="color_counter",
         icon="mdi:file-document-outline",
         name="Color counter",
         native_unit_of_measurement=UNIT_PAGES,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.color_counter,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_DUPLEX_COUNTER,
+        key="duplex_unit_pages_counter",
         icon="mdi:file-document-outline",
         name="Duplex unit pages counter",
         native_unit_of_measurement=UNIT_PAGES,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.duplex_unit_pages_counter,
+        extra_state_attrs=lambda _: {},
     ),
     BrotherSensorEntityDescription(
-        key=ATTR_DRUM_REMAINING_LIFE,
+        key="drum_remaining_life",
         icon="mdi:chart-donut",
         name="Drum remaining life",
         native_unit_of_measurement=PERCENTAGE,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda data: data.drum_remaining_life,
+        extra_state_attrs=lambda data: {
+            ATTR_REMAINING_PAGES: data.drum_remaining_pages,
```

Comment:
I'd move all these attributes to separate sensors.

Reply:
As you wish 😁 https://github.com/home-assistant/core/pull/79932